packages/components/src/unit-control/styles/unit-control-styles.ts

```diff
@@ -106,7 +113,7 @@ const unitSelectSizes = ( { selectSize = 'default' }: SelectProps ) => {
 			justify-content: center;
 			align-items: center;
 
-			&:hover {
+			&:not( :disabled ):hover {
```

Comment:
```suggestion
			&:where( :not( :disabled ) ):hover {
```

I think we need to reduce the specificity here, to prevent the hover style from overriding the focus style when the select is already focused when hovered. (By the way I confirmed that the difference between the hover and focus style is [intentional](https://github.com/WordPress/gutenberg/pull/42000#issuecomment-1168598352).)